java_src/foam/dao/history/HistoryDAO.java

```diff
@@ -69,8 +70,9 @@ public FObject put(FObject obj) {
     FObject current = this.find(obj);
 
     // add new history record
+    String objectId = (String) ((PropertyInfo) obj.getClassInfo().getAxiomByName("objectId")).f(obj);
```

Comment:
Object id = obj.getClassInfo().getId(obj);  (you don't know if it will be a String).